This file is part of MXE.
See index.html for further information.

From 0bed5300536bb6e99a980a56db61aee2a2434e72 Mon Sep 17 00:00:00 2001
From: Timothy Gu <timothygu99@gmail.com>
Date: Tue, 23 Sep 2014 09:36:28 -0700
Subject: [PATCH 1/2] create_cmake: Fix mingw plugin path

MinGW static libs use libfoo.a format, and not foo.lib.
---
 mkspecs/features/create_cmake.prf | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/mkspecs/features/create_cmake.prf b/mkspecs/features/create_cmake.prf
index 2533b7b..4147fe9 100644
--- a/mkspecs/features/create_cmake.prf
+++ b/mkspecs/features/create_cmake.prf
@@ -133,11 +133,19 @@ contains(CONFIG, plugin) {
     CMAKE_PLUGIN_NAME = $$PLUGIN_CLASS_NAME
 
     win32 {
-        isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dll
-        else: CMAKE_PlUGIN_EXT = .lib
+        mingw {
+            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dll
+            else: CMAKE_PlUGIN_EXT = .a
 
-        CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/$${TARGET}$${CMAKE_PlUGIN_EXT}
-        CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/$${TARGET}d$${CMAKE_PlUGIN_EXT}
+            CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/lib$${TARGET}$${CMAKE_PlUGIN_EXT}
+            CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/lib$${TARGET}d$${CMAKE_PlUGIN_EXT}
+        } else {
+            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dll
+            else: CMAKE_PlUGIN_EXT = .lib
+
+            CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/$${TARGET}$${CMAKE_PlUGIN_EXT}
+            CMAKE_PLUGIN_LOCATION_DEBUG = $$PLUGIN_TYPE/$${TARGET}d$${CMAKE_PlUGIN_EXT}
+        }
     } else {
         mac {
             isEmpty(CMAKE_STATIC_TYPE): CMAKE_PlUGIN_EXT = .dylib
-- 
1.8.3.2
